.github/workflows/test-and-lint.yml

```diff
@@ -15,4 +15,27 @@ jobs:
     - name: Install
       run: npm ci
     - name: Test
-      run: npm run test
\ No newline at end of file
+      run: npm run test
+
+  unit-test-with-coverage-and-sonar:
+    runs-on: ubuntu-latest
+    steps:
+    - name: Checkout
+      uses: actions/checkout@v2
+    - name: Set up node
+      uses: actions/setup-node@v1
+      with:
+        node-version: '15.x'
```

Comment:
should the node-version be a variable ? 
would having different node-versions (line 14) be an issue ?

Reply:
Node 10 is severely outdated, but it is what is still used by our docker container, I think. I'm all in favor of upgrading that everywhere to a more recent version.

However, I would do that in a different PR. I chose 15 here mainly because that is what I used in my own project so far and I don't see any downside in using it. If at all, it gives us more confidence for the migration mentioned above.

Reply:
I see. definitely the upgrade should be a different PR.